route/bgmetadata_test.go

```diff
@@ -0,0 +1,71 @@
+package route
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/graphite-ng/carbon-relay-ng/encoding"
+	"github.com/stretchr/testify/assert"
+)
+
+func testBloomFilterConfig() BloomFilterConfig {
+	const (
+		shardingFactor = 3
+		filterSize     = 1000000
+		faultTolerance = 0.0000001
+		clearInterval  = time.Duration(300 * time.Millisecond)
+		cache          = "" // don't test caching
+	)
+	var clearWait time.Duration
+	bfc, _ := NewBloomFilterConfig(
+		filterSize,
+		faultTolerance,
+		shardingFactor,
+		cache,
+		clearInterval,
+		clearWait,
+	)
+	return bfc
+}
+
+func testBgMetadata(t *testing.T) *BgMetadata {
+	const (
+		key    = "test_route"
+		prefix = ""
+		sub    = ""
+		regex  = ""
+	)
+	bfc := testBloomFilterConfig()
+	m, _ := NewBgMetadataRoute(key, prefix, sub, regex, bfc)
+	m.ctx, m.cancel = context.WithCancel(context.Background())
+	return m
+}
+
+func TestMetricFiltering(t *testing.T) {
+	m := testBgMetadata(t)
+	dp := make([]encoding.Datapoint, m.bfCfg.ShardingFactor)
+	dp[0] = encoding.Datapoint{Name: "metric.name.aaaa"}      // 0
+	dp[1] = encoding.Datapoint{Name: "metric.name.aaaaaaaaa"} // 1
+	dp[2] = encoding.Datapoint{Name: "metric.name.bbbb"}      // 2
+
+	// check if right filters are filled
+	for i := 0; i < len(dp); i++ {
+		m.Dispatch(dp[i])
+		assert.True(t, m.shards[i].filter.TestString(dp[i].Name))
+	}
+	m.Dispatch(dp[0])
+
+	// check if added metrics are cleared from the filters
+	go m.clearBloomFilter()
+	time.Sleep(1 * time.Second)
```

Comment:
overall it's a bad habit to time.Sleep in tests, as it slows down every test in the suite 

FAR: try to avoid it

Reply:
I'll see what I can do. One second didn't seem like much. Worst case scenario, I can reduce the time. 👍